mempool/mempool.go

```diff
@@ -29,6 +29,9 @@ type Mempool interface {
 	// transactions (~ all available transactions).
 	ReapMaxTxs(max int) types.Txs
 
+	// GetNextTransaction will return transaction with condition that Bytes and Gas
+	// must be bess than remainBytes and remainGas, and with highest priority less than prior
+	GetNextTransaction(remainByte int64, gas int64, prior []byte) types.Tx
```

Comment:
`prio` is more often used as the acronym for priority. in this case though I think being explicit by writing the actual word doesn't hurt